### PR TITLE
Dapps and RPC server merge

### PR DESCRIPTION
**Pull request by tomusdrw**
Part of #4474 

- Introducing fast minihttp server (around 5x faster on a single thread, with possibility to scale to any number of threads; requires dapps to be disabled)
- Removed separate Dapps server and implemented dapps as RPC middleware
- Removed all dapps-related flags (including dapps authorization that was borked anyway)

### CODE REVIEW

File: dapps/src/tests/rpc.rs [hunk 1]
```diff
@@ -49,70 +47,72 @@ fn should_serve_rpc() {
...
-		",
-		req.as_bytes().len(),
-		req,
-	));
-
-	// then
-	response.assert_status("HTTP/1.1 200 OK");
-	assert_eq!(response.body, "31\n{\"jsonrpc\":\"2.0\",\"result\":\"Hello World!\",\"id\":1}\n\n0\n\n".to_owned());
-}
+// TODO [Move to RPC tests!]
```
Comment (tomusdrw) [inline dapps/src/tests/rpc.rs]: Oops, just noticed the TODO :)

**Comment by gavofyork** [comment]
can't we have the dapps use the "fast minihttp server"?

File: Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,6 @@ serde_json = "0.9"
 app_dirs = "1.1.1"
 fdlimit = "0.1"
 ws2_32-sys = "0.2"
-hyper = { default-features = false, git = "https://github.com/paritytech/hyper" }
```
Comment (debris) [inline Cargo.toml]: 🎉 🎉 🎉

File: parity/deprecated.rs [hunk 3]
```diff
@@ -21,93 +21,88 @@ use cli::Args;
...
-	if args.flag_ipcdisable {
-		result.push(Deprecated::ipcdisable());
+	if args.flag_dapps_user.is_some() {
+		result.push(Deprecated::Removed("--dapps-user"));
 	}
 
-	if args.flag_ipc_off {
-		result.push(Deprecated::ipc_off());
+	if args.flag_dapps_pass.is_some() {
+		result.push(Deprecated::Removed("--dapps-pass"));
```
Comment (debris) [inline parity/deprecated.rs]: How does json-rpc authentication work now? (I was not following your recent prs closely) :p

Comment (debris) [inline parity/deprecated.rs]: nvm, I read pr description

**Comment by tomusdrw** [comment]
@gavofyork That would require quite some work to re-write current dapps stuff (but might be worth considering in the future).
Also `minihttp` is not really full-featured HTTP server, it's missing some stuff like streaming responses in chunks, handling cached requests properly and we rely on this for some parts of dapps server responsibilities (streaming hash-addresed content and caching it, streaming dapps packages, etc.).

**Pull request merged**